utils/monitoring.js

```diff
@@ -0,0 +1,113 @@
+const { errors } = require('arsenal');
+const promClient = require('prom-client');
+const { Registry } = require('prom-client');
+const CountMaster = require('../CountItems/CountMaster');
+
+const aggregatorRegistry = new promClient.AggregatorRegistry();
+const { collectDefaultMetrics } = promClient;
+
+const bucketCount = (promClient.register.getSingleMetric('bucketCount_metrics'))
+    || new promClient.Gauge({
+        name: 'bucketCount_metrics',
+        help: 'Total number of buckets processed',
+        labelNames: ['bucketName'],
+    });
+
+// Histogram of the bucket processing duration, by the utilization service.
+
+const bucketProcessingDuration = (promClient.register.getSingleMetric('count_items_bucketProcessingDuration'))
+    || new promClient.Histogram({
+        name: 'count_items_bucketProcessingDuration',
```

Comment:
naming convention, but also `processing` is overuse ("consolidation" is actually more processing than the first phase)...

```suggestion
        name: 's3_countitems_bucket_listing_duration_seconds',
```